### PR TITLE
scope tls warning suppression

### PR DESCRIPTION
**Pull request by drewmullen**
insecure tls warning was being suppressed but should only be suppressed when `--skip-tls-verify` is passed as a param.

ignoring C901 because of flake error. will reinclude if we breakup the parseargs portion of this module: 
`./kv_recursive.py:124:1: C901 'If 124' is too complex (11)`

### CODE REVIEW

**Comment by drewmullen** [comment]
hmm, may have to adjust codecov targetting

**Comment by drewmullen** [comment]
added `.codecov.yml` which sets the range for reporting unit % status colors. also disable statuses from codecov. we will get comments and have test reporting but they will not affect our PR status
```
$ cat .codecov.yml
coverage:
  range: 40..70
  round: down
  precision: 1
  status:
    project: off
    patch: off
```

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # E128: Line continuation not lined up
 # E402: Imports can be spaced out
 # W503: Making if's readable
-ignore = E121,E128,E402,W503,E12
+ignore = E121,E128,E402,W503,E12,C901
```
Comment (sarcasticadmin) [inline tox.ini:7]: Not sure why this cause `flake8` to complain but this seems fine for the time being

File: kv_recursive.py [hunk 2]
```diff
@@ -160,6 +147,10 @@ def main():
     if not args.destination_token:
         args.destination_token = args.source_token
 
+    if not args.tls_skip_verify:
```
Comment (sarcasticadmin) [inline kv_recursive.py:150]: Much better 👍

File: .codecov.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+coverage:
+  range: 40..70
+  round: down
+  precision: 1
+  status:
```
Comment (sarcasticadmin) [inline .codecov.yml:5]: Thanks for digging into this

**Review by sarcasticadmin — APPROVED** [review]
@drewmullen definitely agree with the approach here for `codecov` and thanks for isolating the tls suppression to only when its been explicitly disabled

File: tox.ini (hunk 1 above)
Comment (drewmullen) [reply tox.ini:7]: so i dont forget, i created an issue to resolve this: https://github.com/drewmullen/vault-kv-migrate/issues/32

its not a priority but at least the use is documented

**Pull request merged**